src/tests/tracing/eventpipe/diagnosticport/diagnosticport.cs

```diff
@@ -0,0 +1,331 @@
+                    Logger.logger.Log($"Sent: {message.ToString()}");
+                    IpcMessage response = IpcClient.SendMessage(ConnectionHelper.GetStandardTransport(pid), message);
+                    Logger.logger.Log($"Received: {response.ToString()}");
+
+                    Logger.logger.Log($"Stopping EventPipeSession");
+                    EventPipeClient.StopTracing(pid, sessionId);
+                    await readerTask;
+                    Logger.logger.Log($"Stopped EventPipeSession");
+
+                    // runtime should have resumed now
+                    fSuccess &= hasResumed;
+                    Logger.logger.Log($"Runtime is resumed (expects: true): {hasResumed}");
+
+                }
+            );
+
+
+            fSuccess &= await subprocessTask;
+
+            return fSuccess;
+        }
+
+        public static async Task<bool> TEST_AdvertiseAndProcessInfoCookiesMatch()
+        {
+            bool fSuccess = true;
+            string serverName = ReverseServer.MakeServerAddress();
+            Logger.logger.Log($"Server name is '{serverName}'");
+            var server = new ReverseServer(serverName);
+            using var memoryStream = new MemoryStream();
+            Task<bool> subprocessTask = Utils.RunSubprocess(
+                currentAssembly: Assembly.GetExecutingAssembly(),
+                environment: new Dictionary<string,string> { { Utils.DiagnosticPortsEnvKey, $"{serverName},connect,nosuspend" } },
+                duringExecution: async (pid) =>
+                {
+                    Stream stream = await server.AcceptAsync();
+                    IpcAdvertise advertise = IpcAdvertise.Parse(stream);
+                    Logger.logger.Log(advertise.ToString());
+
+                    Logger.logger.Log($"Send ProcessInfo Diagnostics IPC Command");
+                    // send ProcessInfo command (0x04=ProcessCommandSet, 0x00=ProcessInfo commandid)
+                    var message = new IpcMessage(0x04,0x00);
+                    Logger.logger.Log($"Sent: {message.ToString()}");
+                    IpcMessage response = IpcClient.SendMessage(stream, message);
+                    Logger.logger.Log($"received: {response.ToString()}");
+                    ProcessInfo info = ProcessInfo.TryParse(response.Payload);
+                    Logger.logger.Log($"ProcessInfo: {{ id={info.ProcessId}, cookie={info.RuntimeCookie}, cmdline={info.Commandline}, OS={info.OS}, arch={info.Arch} }}");
+
+                    Utils.Assert(info.RuntimeCookie.Equals(advertise.RuntimeInstanceCookie), $"The runtime cookie reported by ProcessInfo and Advertise must match.  ProcessInfo: {info.RuntimeCookie.ToString()}, Advertise: {advertise.RuntimeInstanceCookie.ToString()}");
+                    Logger.logger.Log($"ProcessInfo and Advertise Cookies are equal");
+                }
+            );
+
+            fSuccess &= await subprocessTask;
+
+            return fSuccess;
+        }
+
```

Comment:
We should probably add a test(s) that ensure the string parsing is robust. Stuff like:
0 length ports, invalid paths, invalid tags, multiple conflicting tags, not setting any tags to use defaults, a path whose name is identical to a tag name

Reply:
For conflicting tag, the logic will use that last tag value specified (which should be tested as you're suggesting).  Would expect conflicting or duplicate tags to produce an error?

Reply:
Added a test that tries various pathological and nonsensical values for the port configs.